arm64: defconfig: Experimental PREEMPT_RT defconfig for BCM2711 + BCM2712)

timg236 opened the pull request:
Github CI workflow test for rpi-update

pelwell commented:
This looks OK, but we'll need to be careful that the new file is created to match the latest arm64/bcm2711_defconfig - this version is already missing three options:
```
< CONFIG_GPIO_FXL6408=m
< CONFIG_DRM_PANEL_ILITEK_ILI9806E_SPI=m
< CONFIG_SND_PISOUND_MICRO=m
```

timg236 commented:
> This looks OK, but we'll need to be careful that the new file is created to match the latest arm64/bcm2711_defconfig - this version is already missing three options:
> 
> ```
> < CONFIG_GPIO_FXL6408=m
> < CONFIG_DRM_PANEL_ILITEK_ILI9806E_SPI=m
> < CONFIG_SND_PISOUND_MICRO=m
> ```

Yes, before this is merged we need to make sure it's in sync and stays in sync. Just using it to run a few normal / RT benchmarks and check for anything weird right now.

pelwell commented on arch/arm64/configs/bcm2711_rt_defconfig:
```diff
@@ -0,0 +1,1735 @@
+CONFIG_LOCALVERSION="v8-rt"
```
I think this needs a leading hyphen, otherwise you get `6.12.19v8-rt+`.

timg236 replied:
Good point. I lost the hyphen when I added a script to generate this for testing. Will fix, I need to test on Pi4 before merging this and offering it as an experimental build.

timg236 committed:
arm64: config: Experimental PREEMPT_RT defconfig

Add a new PREEMPT_RT arm64 config based on the bcm2711_defconfig
for BCM2711 and BCM2712.

Signed-off-by: Tim Gover <tim.gover@raspberrypi.com>

pelwell commented on .github/workflows/kernel-build.yml:
```diff
@@ -50,6 +50,11 @@ jobs:
 
+          - name: bcm2711_rt
+            arch: arm64
+            defconfig: bcm2711_defconfig
```
bcm2711_rt_defconfig, surely?

pelwell commented on .github/workflows/kernel-build.yml:
```diff
@@ -50,6 +50,11 @@ jobs:
+          - name: bcm2711_rt
+            arch: arm64
+            defconfig: bcm2711_defconfig
+            kernel: kernel8
```
Is it not better to use a different name?

timg236 committed:
workflows: Add arm64 bcm2711_rt build

Add a Github CI workflow bcm2711_rt_defconfig

Signed-off-by: Tim Gover <tim.gover@raspberrypi.com>
